docs/en/advanced_guides/models.md

```diff
@@ -0,0 +1,330 @@
+        noise_batch = noise_batch.to(get_module_device(self))
+        x = self.noise2feat(noise_batch)
+        x = self.upsampling(x)
+        x = self.output_layer(x)
+        return x
+```
+
+If you want to implement specific weights initialization method for you network, you need add `init_weights` function by yourself.
+
+```python
+    def init_weights(self, pretrained=None):
+        if isinstance(pretrained, str):
+            logger = MMLogger.get_current_instance()
+            load_checkpoint(self, pretrained, strict=False, logger=logger)
+        elif pretrained is None:
+            for m in self.modules():
+                if isinstance(m, (nn.Conv2d, nn.ConvTranspose2d)):
+                    normal_init(m, 0, 0.02)
+                elif isinstance(m, _BatchNorm):
+                    nn.init.normal_(m.weight.data)
+                    nn.init.constant_(m.bias.data, 0)
+        else:
+            raise TypeError('pretrained must be a str or None but'
+                            f' got {type(pretrained)} instead.')
+```
+
+After the implementation of class `DCGANGenerator`, we need to update the model list in `mmgen/models/architectures/__init__.py`, so that we can import and use class `DCGANGenerator` by `mmgen.models.architectures`.
+
+Implementation of Class `DCGANDiscriminator` follows the similar logic, and you can find the implementation [here](https://github.com/open-mmlab/mmgeneration/blob/test-1.x/mmgen/models/architectures/dcgan/generator_discriminator.py#L195).
+
+## Step 2: Define the your **Model**
+
+After the implementation of the network **Architecture**, we need to define our **Model** class `DCGAN`.
+
+In **Model**, you should implement three function `train_step`, `val_step` and `test_step`.
+
+- `train_step`: This function is responsible to update the parameters of the network and called by `IterBasedTrainLoop` or `EpochBasedTrainLoop`. `train_step` take data batch as input and return a dict of log.
+- `val_step`: This function is responsible for getting output for validation during the training process. and is called by `GenValLoop`.
+- `test_step`: This function is responsible for getting output in test process and is called by `GenTestLoop`.
+
+For simplify using, we provide `BaseGAN` class im MMGeneration, which implement generic `train_step`, `val_step` and `test_step` function for GAN models. With `BaseGAN`, each specific GAN algorithm only need to implement `train_generator` and `train_discriminator`.
+
+In `train_step`, we support data preprocessing, gradient accumulation (realized by [`OptimWrapper`](https://github.com/open-mmlab/mmengine/blob/main/docs/en/tutorials/optim_wrapper.md)) and expontial moving averate (EMA) realized by [(`ExponentialMovingAverage`)](https://github.com/open-mmlab/mmgeneration/blob/test-1.x/mmgen/models/averaged_model.py#L19). With `BaseGAN.train_step`, each specific GAN algorithm only need to implement `train_generator` and `train_discriminator`.
```

Comment:
```suggestion
In `train_step`, we support data preprocessing, gradient accumulation (realized by [`OptimWrapper`](https://github.com/open-mmlab/mmengine/blob/main/docs/en/tutorials/optim_wrapper.md)) and expontial moving averate (EMA) realized by [(`ExponentialMovingAverage`)](https://github.com/open-mmlab/mmgeneration/blob/dev-1.x/mmgen/models/averaged_model.py#L19). With `BaseGAN.train_step`, each specific GAN algorithm only need to implement `train_generator` and `train_discriminator`.
```

Reply:
seems we have no introduction for data preprocessing in whole documents, we may give its introduction here.